.github/workflows/ci.yml

```diff
@@ -26,6 +26,15 @@ jobs:
           python -m pip install --disable-pip-version-check .
       - name: Run tests on ${{ matrix.os }}
         run: nox --non-interactive --error-on-missing-interpreter --session "tests-${{ matrix.python-version }}" -- --full-trace
+  action-default-tests:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-18.04, ubuntu-20.04, windows-2019, windows-2022, macos-10.15, macos-11, macos-12]
```

Comment:
Small nitpick but could you change e.g. `windows-2022` to `windows-latest`, likewise for `ubuntu-20.04` and `macos-11`. Just means less changes when things update

Reply:
The goal here is to test on every GitHub-hosted runner available.
Given that goal, I think it's best expressed using their qualified names. Yes there's a (small) price to pay when runners are added and/or removed but that's probably the only way to ensure the nox action is working correctly on all available runners.

PS: `macos-latest` is (for now) `macos-11` and it just seems strange to have `macos-latest` to reference `macos-11` when `macos-12` is also present. The same will happen with `ubuntu-22.04`/`ubuntu-20.04`/`ubuntu-latest` in a near future https://github.com/actions/virtual-environments/issues/5428

Reply:
Fair enough, I suppose we should be testing against every runner. FYI the `-latest` is the latest _stable_ release of the OS, so macos-12 (Monterey) is in public beta

Reply:
added ubuntu 22.04 that's now in public beta: https://github.com/actions/virtual-environments/issues/5490